GoogleSignIn/Sources/GIDGoogleUser.m

```diff
@@ -99,21 +121,43 @@ - (nullable instancetype)initWithCoder:(NSCoder *)decoder {
     _authentication = [decoder decodeObjectOfClass:[GIDAuthentication class]
                                             forKey:kAuthenticationKey];
     _grantedScopes = [decoder decodeObjectOfClass:[NSArray class] forKey:kGrantedScopesKey];
-    _userID = [decoder decodeObjectOfClass:[NSString class] forKey:kUserIDKey];
-    _serverAuthCode = [decoder decodeObjectOfClass:[NSString class] forKey:kServerAuthCodeKey];
     _profile = [decoder decodeObjectOfClass:[GIDProfileData class] forKey:kProfileDataKey];
-    _hostedDomain = [decoder decodeObjectOfClass:[NSString class] forKey:kHostedDomainKey];
+    _authState = [decoder decodeObjectOfClass:[OIDAuthState class] forKey:kAuthState];
+    _idToken = [decoder decodeObjectOfClass:[NSString class] forKey:kIDToken];
   }
   return self;
 }
 
 - (void)encodeWithCoder:(NSCoder *)encoder {
   [encoder encodeObject:_authentication forKey:kAuthenticationKey];
   [encoder encodeObject:_grantedScopes forKey:kGrantedScopesKey];
-  [encoder encodeObject:_userID forKey:kUserIDKey];
-  [encoder encodeObject:_serverAuthCode forKey:kServerAuthCodeKey];
   [encoder encodeObject:_profile forKey:kProfileDataKey];
-  [encoder encodeObject:_hostedDomain forKey:kHostedDomainKey];
+  [encoder encodeObject:_authState forKey:kAuthState];
+  [encoder encodeObject:_idToken forKey:kIDToken];
+}
+
+#pragma mark - private method
```

Comment:
I'd name this section "Private Methods" and move it above the protocol implementations.

Reply:
Normally we put the private sections in the end, does it?